src/ducks/auth.ts

```diff
@@ -1,72 +1,198 @@
+import { NavigationContainerRef } from "@react-navigation/native";
 import { encode as base64Encode } from "@stablelib/base64";
 import { Networks } from "@stellar/stellar-sdk";
 import { KeyType, ScryptEncrypter } from "@stellar/typescript-wallet-sdk-km";
 import {
+  HASH_KEY_EXPIRATION_MS,
   NETWORKS,
   SENSITIVE_STORAGE_KEYS,
   STORAGE_KEYS,
 } from "config/constants";
 import { logger } from "config/logger";
-import { Account, HashKey, KeyPair, TemporaryStore } from "config/types";
+import { ROOT_NAVIGATOR_ROUTES, RootStackParamList } from "config/routes";
+import {
+  Account,
+  AUTH_STATUS,
+  AuthStatus,
+  HashKey,
+  KeyPair,
+  TemporaryStore,
+} from "config/types";
 import {
   deriveKeyFromPassword,
   encryptDataWithPassword,
   generateSalt,
+  decryptDataWithPassword,
 } from "helpers/encryptPassword";
 import { createKeyManager } from "helpers/keyManager/keyManager";
-import { isHashKeyValid } from "hooks/useGetActiveAccount";
+import { isHashKeyExpired } from "hooks/useGetActiveAccount";
 import { t } from "i18next";
+import { clearTemporaryData, getHashKey } from "services/storage/helpers";
 import {
   dataStorage,
   secureDataStorage,
 } from "services/storage/storageFactory";
-import { fromMnemonic } from "stellar-hd-wallet";
+import StellarHDWallet from "stellar-hd-wallet";
```

Comment:
@Brunonascdev let's try doing it in a separate PR, but I think we should update this package.
It's been recently upgraded from `0.0.10` to `1.0.1` a few days ago after 6 years of inactivity 😅 
As a reference we have this [recent task](https://github.com/orgs/stellar/projects/58/views/2?pane=issue&itemId=104072709&issue=stellar%7Cfreighter%7C1933) for the extension.

Reply:
I've created an issue for us to tackle this next sprint: https://github.com/stellar/freighter-mobile/issues/76